sdk/clientcore/core/src/main/java/io/clientcore/core/instrumentation/logging/LoggingEvent.java

```diff
@@ -305,30 +300,72 @@ private String getMessageWithContext(String message) {
                 jsonWriter.writeStringField("message", message);
             }
 
-            if (globalPairs != null) {
-                for (Map.Entry<String, Object> kvp : globalPairs.entrySet()) {
-                    jsonWriter.writeUntypedField(kvp.getKey(), kvp.getValue());
-                }
-            }
+            writeContext(jsonWriter);
 
-            if (keyValuePairs != null) {
-                for (Map.Entry<String, Object> kvp : keyValuePairs.entrySet()) {
-                    jsonWriter.writeUntypedField(kvp.getKey(), kvp.getValue());
-                }
-            }
+            jsonWriter.writeEndObject().flush();
 
-            if (eventName != null) {
-                jsonWriter.writeStringField(EVENT_NAME_KEY, eventName);
-            }
+            return outputStream.toString(StandardCharsets.UTF_8);
+        } catch (IOException ex) {
+            throw new UncheckedIOException(ex);
+        }
+    }
+
+    void recordExceptionDetails(Throwable throwable, String message) {
+        addKeyValueInternal(EXCEPTION_TYPE_KEY, throwable.getClass().getCanonicalName());
+        if (message != null) {
+            addKeyValueInternal(EXCEPTION_MESSAGE_KEY, message);
+        }
+
+        if (logger != null && logger.canLogAtLevel(LogLevel.VERBOSE)) {
+            StringBuilder stackTrace = new StringBuilder();
+            DefaultLogger.appendThrowable(stackTrace, throwable);
+            addKeyValue(EXCEPTION_STACKTRACE_KEY, stackTrace.toString());
+        }
+    }
+
+    String getExceptionMessageWithContext(String shortMessage, Throwable cause) {
```

Comment:
Do we want to add in suppressed exception handling too?

Reply:
I don't think suppression should be in the message itself, users can access it and log suppressed exceptions if they want to. We can do a better job logging them to.

Do you have something specific in mind?

Reply:
Nope, I'm fine if we're happy with the behavior and call it out